profiling/src/profile/mod.rs

```diff
@@ -400,12 +400,14 @@ impl Profile {
     ///                may also accidentally pass an earlier time. The duration will be set to zero
     ///                these cases.
     pub fn serialize(
-        &self,
+        self,
         end_time: Option<SystemTime>,
         duration: Option<Duration>,
     ) -> anyhow::Result<EncodedProfile> {
         let end = end_time.unwrap_or_else(SystemTime::now);
         let start = self.start_time;
+        let endpoints_stats = self.endpoints.stats.clone();
```

Comment:
Why do these need cloned since we have it by `self`?

Reply:
We fully consume `self` on the next line, so we can't partially consume it here.  But `std::mem::take` would work, and save the clone.